src/main/java/javax/security/enterprise/CallerPrincipal.java

```diff
@@ -16,13 +16,14 @@
 
 package javax.security.enterprise;
 
+import java.io.Serializable;
 import java.security.Principal;
 
 /**
  * Principal that represents the caller principal associated with the invocation being
  * processed by the container (e.g. the current HTTP request).
  */
-public class CallerPrincipal implements Principal {
+public class CallerPrincipal implements Principal, Serializable {
 
```

Comment:
```suggestion

    private static final long serialVersionUID = 1;

```

Reply:
Didn't notice this was missing. In order to be consistent with other specs, we should use the generated number.

Anyway I'm personally not a fan of hardcoded servial version UIDs as you need to update them every time you change the class (otherwise they don't serve their purpose). What do you think?

Reply:
I'm not a super fan either, but everything from IDEs to maven code analysers to Sonar flags it as warning if you don't have one.

Reply:
Good call, I've added a generated UID.